tools/stack_decode.py

```diff
@@ -10,6 +10,11 @@
 #
 # In each case this script will add file and line information to any backtrace log
 # lines found and echo back all non-Backtrace lines untouched.
+#
+# This has been found to work best if the envoy binary was built with gcc, and with
```

Comment:
This script is for use in production where these conditions are unlikely. Does it work with clang and unstripped binary?

Reply:
I was not able to get *anything* to work for better stack tracing with a standard container-built unstripped binary. stack_decode.py did not produce accurate results before, and did not produce accurate results after, this change, on such a binary.

The behavior should be effectively unchanged (or, if anything, slightly more likely to work for the same reasons as it helps gcc) after this change.

The comment suggesting gcc is specifically to save people the two days of messing with this trying to get it to do anything useful in an environment where it will not. If someone more versed in getting clang debugging to work would like to explain how I would very much welcome it. :)

(My suspicion is that perhaps the container's versions of debugging tools don't match the container's version of the build tools, based on the dwarf addressing errors, which look like `addr2line: DWARF error: invalid or unhandled FORM value: 0x25`.)

Thread discussing this [here](https://envoyproxy.slack.com/archives/G79D81J7Q/p1711046977667749).